c/omni_lock.c

```diff
@@ -396,8 +398,22 @@ int main() {
   // args (args.id)
   CkbIdentityType identity = {0};
 
-  err = parse_witness_lock(&witness_lock);
+  mol2_cursor_t seal = {0};
+  /*
+   * When it fails, WitnessArgs is used. No cobuild enabled.
+   */
+  err = ckb_parse_message(g_cobuild_signing_message_hash, &seal);
+  if (err) {
+    printf("cobuild disabled");
+    g_cobuild_enabled = false;
+    err = parse_witness_lock(&witness_lock, NULL);
+  } else {
+    printf("cobuild enabled");
+    g_cobuild_enabled = true;
+    err = parse_witness_lock(&witness_lock, &seal);
```

Comment:
If we want to move forward with this change, I would suggest that a prefix byte should be added at least here. The prefix byte can then be used to control how the signing message is generated(much like the algorithm id in script args controlling signing algorithm to use). Right now the prefix byte might accept only one value that represents the blake2b based message calculation flow. But in the future, this very prefix byte, can then be used to introduce more variations on signing message calculation, such as a structured text based signing message

Reply:
So we're going to implement it below, without molecule encoding:
```text
<message calculation flow, 1 byte> <seal part>
```